### DISCUSSION TITLE
New fulfillment - draft order additional field

### DISCUSSION THREAD

**Discussion by ikkecode** [medusajs/medusa#5819] [Q&A]
Hello, I'm currently developing a new fulfillment provider, hopefully set to launch before the new year. 

While I've made significant progress in understanding Medusa's workings and overall flow, I'm facing a challenge related to admin extensibility.

3rd party service that my fulfillment integrates with requires data that user needs to pass.

I've chosen to store the required data in the data property of the shipping method. My concern now is how to seamlessly extend the admin draft order UI to incorporate an additional field for the shop owner to input this data.

I'm looking for advice on extending the admin draft order UI to add a field and ensure seamless data storage in the shipping method's data property, Without making it collide with future Medusa versions and other plugins

**Comment by srindom** [comment]
First and foremost your approach with storing the user data in the ShippingMethod's data field is correct, and I agree that it would make sense to be able to manage that data from Admin too.

Admin extensions should be done through the use of Widgets or UI Routes - however, we don't currently support extensions on draft orders. This unfortunately means that it would require a deeper customization that cannot guarantee conflicts-free updates.
